### PR TITLE
660 review select searchbar tests

### PR DESCRIPTION
**Pull request by GCHQ-Developer-741**
## Summary of the changes
Review and changes to the e2e and unit tests for `ic-select` and `ic-search-bar`

## Related issue
#660 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

File: packages/web-components/src/components/ic-select/ic-select.spec.tsx [hunk 1]
```diff
@@ -186,248 +169,642 @@ describe("ic-select", () => {
...
+    await page.rootInstance.handleKeyDown({
+      key: "Backspace",
+      preventDefault: (): void => null,
+    });
+    await page.waitForChanges();
+
+    expect(page.rootInstance.setFocus).toHaveBeenCalledTimes(3);
+  });
+
+  it("should test menu handleKeyboardOpen method - arrow up", async () => {
```
Comment (gd2910) [inline packages/web-components/src/components/ic-select/ic-select.spec.tsx:747]: Isolated menu tests have been created in the ic-menu.spec, I don't see that menu in conjunction with select does any harm but could possibly be removed if considered redundant?

**Review by gd2910 — DISMISSED** [review]

**Review by gd2910 — APPROVED** [review]

File: packages/web-components/src/components/ic-select/ic-select.e2e.ts [hunk 2]
```diff
@@ -1053,6 +1133,33 @@ describe("ic-select", () => {
       expect(menuOptions[3]).toEqualText("Macchiato");
     });
 
+    //TODO: Uncomment when bug is fixed
```
Comment (ad9242) [inline packages/web-components/src/components/ic-select/ic-select.e2e.ts:1136]: has this bug been fixed, or is it a separate ticket?

File: packages/web-components/src/components/ic-select/ic-select.e2e.ts [hunk 3]
```diff
@@ -1910,6 +2106,46 @@ describe("ic-select", () => {
       </ic-input-container>`);
     });
 
+    it("should render a native select when searchable is applied on mobile", async () => {
+      await page.setContent(getTestSearchableSelect(options));
+      await page.waitForChanges();
+
+      const select = await page.find("ic-select");
+
+      expect(select.shadowRoot)
```
Comment (ad9242) [inline packages/web-components/src/components/ic-select/ic-select.e2e.ts:2115]: could this be moved into the snapshot file? - there is a separate one for e2e tests

**Review by ad9242 — CHANGES_REQUESTED** [review]

File: packages/web-components/src/components/ic-select/ic-select.e2e.ts (hunk 2 above)
Comment (GCHQ-Developer-741) [reply packages/web-components/src/components/ic-select/ic-select.e2e.ts:1136]: This is a seperate ticket that has been pulled into sprint (https://github.com/mi6/ic-ui-kit/issues/704). I left a note on that issue telling whoever picks this up to uncomment this when that ticket is done

**Review by gd2910 — APPROVED** [review]

**Review by ad9242 — APPROVED** [review]

**Pull request merged**